### PR TITLE
HSEARCH-2736 Discovery_scheme not taken in account

### PR DESCRIPTION
**Pull request by yodamad**
Fix deprecation on flag discovery_scheme for https support

### CODE REVIEW

**Comment by yrodiere** [comment]
ok to test

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java [hunk 1]
```diff
@@ -81,16 +84,26 @@ private Sniffer createSniffer(String scopeName, RestClient client, Properties pr
...
-					)
-					.build();
+					);
+			String scheme = ConfigurationParseHelper.getString(properties, propertyPrefix + ElasticsearchEnvironment.DISCOVERY_SCHEME, "http");
+
+			// https discovery support
+			if ( scheme.equals(ElasticsearchHostsSniffer.Scheme.HTTPS.toString()) ) {
+				HostsSniffer hostsSniffer = new ElasticsearchHostsSniffer(
+						client,
+						ElasticsearchEnvironment.Defaults.DISCOVERY_REFRESH_INTERVAL * 1_000,
```
Comment (yrodiere) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java]: Unless I'm mistaken, this should be `ElasticsearchHostsSniffer.DEFAULT_SNIFF_REQUEST_TIMEOUT`.

**Comment by Sanne** [comment]
could one of you add a test as well? I'd hate this to have regressions in the future.

If noone has time for that right now I'm happy to merge this but let's open a JIRA to make sure a test is added in this development cycle.

**Comment by yodamad** [comment]
If possible, i'd like you to merge first because it's really blocking me to complete integration of elasticsearch 5.4 and hibernatesearch in a fully "secured" environment. 
I'll have a look for the test asap, but I first need to see & understand how your tests work.
Can you also tell me what kind of test you'd like me to add ?

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java (hunk 1 above)
Comment (yodamad) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java]: Do I have to do another PR after ammending ?

Comment (Sanne) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java]: as you prefer. Typically we just force-push on the same branch, that will replace the current code in the same PR.

**Comment by Sanne** [comment]
> If possible, i'd like you to merge first because it's really blocking me 

Sure that's fine. Also the test is possibly complex in this case, and @yrodiere seems to have some ideas already. I'd normally ask contributors for a test but this is a complex one, let's leave it to him.

**Comment by yrodiere** [comment]
@yodamad I'm working on the test right now

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java (hunk 1 above)
Comment (yodamad) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java]: I've just ammend commit (https://github.com/yodamad/hibernate-search/commit/3538cacf0b7d735b8ccdaac4611a6fe9ff4616ec).

**Comment by yrodiere** [comment]
@yodamad Still on it, it will require some hacking... In the meantime, could you please ping me when you signed the CLA? Thanks.

**Comment by yodamad** [comment]
@yrodiere I've signed it (mvincent.it@gmail.com)

**Comment by yrodiere** [comment]
@Sanne: I managed to set up a test, but due to [HSEARCH-2481](https://hibernate.atlassian.net/browse/HSEARCH-2481) it won't work on CI. It does work locally though (fails before the commit, passes after the commit), so I added it anyway but disabled it. We'll have to take a look at HSEARCH-2481.

@yodamad Thanks! I merged the fix, and a 5.8-SNAPSHOT version with this fix will be available tomorrow (after the nightly build).

**Comment by Sanne** [comment]
great, thanks all!

**Comment by yrodiere** [comment]
> a 5.8-SNAPSHOT version with this fix will be available tomorrow (after the nightly build).

On a side note, in order to use snpashots, you must use the JBoss Maven repository. See [settings-example.xml](https://github.com/hibernate/hibernate-search/blob/master/settings-example.xml) for an example of how to do that.
